### DISCUSSION TITLE
How to set DataTemplates to static instance

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#10504] [General]
What is the syntax to take the DataTemplates from an existing static instance? To define something like this...

    DataTemplates="{x:Static local:App.MainWindow}"

but how do I set it in App.axaml?

```xaml
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:local="clr-namespace:Demo.Avalonia.StrongLocator"
             x:Class="Demo.Avalonia.StrongLocator.App">
    <Application.DataTemplates>
        <local:App.ViewLocator/>
    </Application.DataTemplates>
```

This syntax gives

    App.axaml(6, 10): [AVLN:0004] Dots aren't allowed in type names Line 6, position 10.

**Comment by maxkatz6** [comment]
`<x:Static Member='local:App.MainWindow' />` from the xaml standard could work, if it was implemented in the compiler.

**Reply by mysteryx93** [reply]
Hum. That's interesting. Rider tells an error at `Member`, but the app starts...

then throws

> System.NullReferenceException: Object reference not set to an instance of an object.
   at Avalonia.Controls.Templates.DataTemplateExtensions.FindDataTemplate(Control control, Object data, IDataTemplate primary) in /_/src/Avalonia.Controls/Templates/DataTemplateExtensions.cs:line 54

**Comment by mysteryx93** [comment]
Is there any solution to this, or should I make the StrongViewLocator internal dictionary static instead?
